OKR/PACKAGE-MANAGERS.md

```diff
@@ -2,6 +2,224 @@
+
+* Cluster has been using badger for a while and their only complaint is memory usage.
+  * We may need to dynamically/automatically tune badger's up-front memory usage.
+* We still have a few remaining issues to fix:
+  * Make sure GC actually garbage collects: https://github.com/ipfs/go-ds-badger/issues/54
+  * Investigate Windows crash/recovery issue reported by @djdv.
+  * Implement periodic datastore-level garbage collection: https://github.com/ipfs/go-ds-badger/issues/51
+
+#### Alternatives
+
+These are some alternative initiatives that _could_ have addressed the KRs.
+
+* Add/Update: A stand-alone IPFS "dropbox" tool. We could write an entirely new tool that simply makes a directory available to the IPFS network using the filestore. Files would be automatically added/removed from IPFS as they're added/removed from the directory.
+  * Pro: Many users would love this kind of tool.
+  * Con: Could be a _lot_ of work.
+  * Con: We want mount support and metadata _anyways_.
+* Update: Out-of-band metadata: Instead of storing timestamps in files to enable rsync, we could store them out-of-band.
+  * Pro: Better deduplication.
+  * Con: Out-of-band data can confuse users (it won't be replicated along with the files).
+  * Con: We need to store some metadata (modes, etc.) anyways.
+  * Con: Some package managers expect files to have accurate timestamps (TODO: check this).
+  * Meh: We're planning on storing metadata in directories, not files. Directories are likely to be unique.
+* Update: A special-purpose rsync tool: We could add a special-purpose `ipfs rsync` command to sync a directory in IPFS with a directory outside of IPFS.
+  * Pro: Pretty simple to implement.
+  * Con: Unlike mount, the files would need to live in IPFS and outside IPFS.
+  * Con: To make this work without timestamps inside IPFS, we'd need the user to pass some kind of "add if modified since" date. That's pretty poor UX.
+
+### Objective: Engage with the package manager community to demonstrate how IPFS meets their needs.
+
+While simply working towards our goal to integrate IPFS and package managers will help drive improvements to the protocol and core implementations, the end goal is to actually _integrate_ IPFS and package managers. To that end, we can't stop at improving the core implementations; we need to demonstrate those improvements to the community.
+
+#### Key Results
+
+The key results for this objective mirror the top three key results in the previous objective.
+
+##### P1: Update Performance
+
+Communicate update performance improvements to the community.
+
+##### P1: Communicate fetch performance improvements
+
+Communicate fetch performance improvements to the community.
+
+##### P3: Communicate add performance improvements
+
+Communicate add performance improvements to the community.
+
+#### Initiatives
+
+In each of these initiatives, we plan to pair with a user/community member.
+
+##### Demo: OpenSuse + Mount
```

Comment:
Question whether RPM is the right tool here - if use rsync likely a good example of where we can get the benefit. Arch sufficiently small that fits nicely on a hard drive...

Reply:
@djdv for thoughts?

Reply:
During a meeting today we discussed the expected trajectory.
We'll aim for openSUSE/RPM but may or may not experiment with other repos/formats as a means of ramping up. Starting with locally made smaller repos for practicality and working up to mirroring actual (large) sets.

Going from 1 target to another seems less reasonable so we'll likely remain with RPM the whole way through.